clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -1714,6 +1714,8 @@ class ConstraintRefersToContainingTemplateChecker
   // Friend, likely because it was referred to without its template arguments.
   void CheckIfContainingRecord(const CXXRecordDecl *CheckingRD) {
     CheckingRD = CheckingRD->getMostRecentDecl();
+    if (!CheckingRD->isTemplated())
```

Comment:
What do you think about this example, which exercises this issue in the above case? https://godbolt.org/z/Yz1WEqM7M

If I switch from `isTemplated()` to `getDescribedTemplate()` to only check `CheckingRD` itself, then clang will accept that code. Otherwise, as the patch is now, the example is rejected. What should be the correct behavior?

Reply:
Hmm... I THINK that should be rejected, since you can't name `Child` without naming `Parent`?  But I don't think the standard is explicit about that.

So perhaps `isTemplated` is what we want here.

I didn't think it too far through when making the comment, but I'm glad you put together a good example.